src/tooltip/tooltip.less

```diff
@@ -1,145 +1,145 @@
-@tooltip-prefix-cls: rc-tooltip;
-
-@font-size-base: 12px;
-@line-height-base: 1.5;
-@border-radius-base: 6px;
-@overlay-shadow: 0 0 4px rgba(0, 0, 0, 0.17);
+@font-size-base: 14px;
 @tooltip-color: @white;
-@tooltip-bg: @gray-coal;
-
-@tooltip-arrow-width: 5px;
-@tooltip-distance: (@tooltip-arrow-width + 4px);
+@tooltip-bg: @gray-lighter;
 @tooltip-arrow-color: @tooltip-bg;
 
-.@{tooltip-prefix-cls} {
-    position: absolute;
-    z-index: 1070;
-    display: block;
-    font-size: @font-size-base;
-    line-height: @line-height-base;
-
-    max-width: 300px;
-
-    &-hidden {
-        display: none;
-    }
-
-    &-placement-top, &-placement-topLeft, &-placement-topRight {
-        padding: @tooltip-arrow-width 0 @tooltip-distance 0;
-    }
-    &-placement-right, &-placement-rightTop, &-placement-rightBottom {
-        padding: 0 @tooltip-arrow-width 0 @tooltip-distance;
-    }
-    &-placement-bottom, &-placement-bottomLeft, &-placement-bottomRight {
-        padding: @tooltip-distance 0 @tooltip-arrow-width 0;
-    }
-    &-placement-left, &-placement-leftTop, &-placement-leftBottom {
-        padding: 0 @tooltip-distance 0 @tooltip-arrow-width;
-    }
-}
-
-.@{tooltip-prefix-cls}-inner {
-    padding: 8px 10px;
-    color: @tooltip-color;
-    text-align: left;
-    text-decoration: none;
+@tooltip-arrow-background-after: @tooltip-bg;
+@tooltip-arrow-background-before: @gray-mid-light;
+
+.tooltip-container {
     background-color: @tooltip-bg;
-    border-radius: @border-radius-base;
-    box-shadow: @overlay-shadow;
+    border-radius: 0px;
+    font-size: @font-size-base;
+    padding: 10px 16px;
+    border: 1px solid @gray-mid;
+    box-shadow: 0 2px 4px @gray-mid;
+    display: flex;
+    flex-direction: column;
+    transition: opacity 0.3s;
+    z-index: 2147483647;
```

Comment:
That is quite a lot :) We should have a proper concept for z-indices. Anyway wouldn't block your PR because of that.